components/net/http_cache.rs

```diff
@@ -428,12 +433,31 @@ fn handle_range_request(
                     },
                     _ => continue,
                 };
-                if res_beginning - 1 < beginning && res_end + 1 > end {
+                let (res_beginning_minus_one, res_end_plus_one) = {
+                    match (res_beginning.checked_sub(1), res_end.checked_add(1)) {
+                        (Some(minus_one), Some(plus_one)) => (minus_one, plus_one),
+                        _ => continue,
+                    }
+                };
+                if res_beginning_minus_one < beginning && res_end_plus_one > end {
```

Comment:
Instead of this block, can't we do `if res_beginning <= beginning && res_end >= end`?